quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/transaction/TransactionalSessionHandler.java

```diff
@@ -1,57 +0,0 @@
-/*
- * Copyright 2022 Red Hat, Inc. and/or its affiliates
- * and other contributors as indicated by the @author tags.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.keycloak.quarkus.runtime.transaction;
-
-import org.keycloak.models.KeycloakSession;
-import org.keycloak.models.KeycloakSessionFactory;
-import org.keycloak.models.KeycloakTransactionManager;
-import org.keycloak.quarkus.runtime.integration.QuarkusKeycloakSessionFactory;
-import org.keycloak.services.DefaultKeycloakSession;
-
-/**
- * <p>A {@link TransactionalSessionHandler} is responsible for managing transaction sessions and its lifecycle. Its subtypes
- * are usually related to components available from the underlying stack that runs on top of the request processing chain
- * as well as at the end in order to create transaction sessions and close them accordingly, respectively.
- */
-public interface TransactionalSessionHandler {
```

Comment:
I was more talking about why removing this interface as it has a purpose and does not seem related to all the nice changes you are introducing here.

I would keep it for now and have a better discussion in a separate issue/PR if we really want to do so. I know what you are doing with this interface but it was there to help identify the components involved in the transaction lifecycle.

That is the kinda of discussion I would like to avoid now and focus on the key bits you are delivering here.

Reply:
> I was more talking about why removing this interface as it has a purpose and does not seem related to all the nice changes you are introducing here.

It's just to move the logic closer together as those are the only places those methods are used.

> That is the kinda of discussion I would like to avoid now and focus on the key bits you are delivering here.

No problem, I've put back TransactionalSessionHandler.